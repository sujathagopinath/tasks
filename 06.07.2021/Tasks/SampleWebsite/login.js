let parameters = {
    count: false,
    letters: false,
    numbers: false,
    special: false
}
let strengthBar = document.getElementById("strength-bar");
let msg = document.getElementById("msg");

function strengthChecker() {
    let password = document.getElementById("password").value;

    parameters.letters = (/[A-Za-z]+/.test(password)) ? true : false;
    parameters.numbers = (/[0-9]+/.test(password)) ? true : false;
    parameters.special = (/[!\"$%&/()=?@~`\\.\';:+=^*_-]+/.test(password)) ? true : false;
    parameters.count = (password.length > 7) ? true : false;

    let barLength = Object.values(parameters).filter(value => value);

    console.log(Object.values(parameters), barLength);

    strengthBar.innerHTML = "";
    for (let i in barLength) {
        let span = document.createElement("span");
        span.classList.add("strength");
        strengthBar.appendChild(span);
    }

    let spanRef = document.getElementsByClassName("strength");
    for (let i = 0; i < spanRef.length; i++) {
        switch (spanRef.length - 1) {
            case 0:
                spanRef[i].style.background = "#ff3e36";
                msg.textContent = "Your password is very weak";
                break;
            case 1:
                spanRef[i].style.background = "#ff691f";
                msg.textContent = "Your password is weak";
                break;
            case 2:
                spanRef[i].style.background = "#ffda36";
                msg.textContent = "Your password is good";
                break;
            case 3:
                spanRef[i].style.background = "#0be881";
                msg.textContent = "Your password is strong";
                break;
        }
    }
}


function toggle() {
    let password = document.getElementById("password");
    let eye = document.getElementById("toggle");

    if (password.getAttribute("type") == "password") {
        password.setAttribute("type", "text");
        eye.style.color = "#0be881";
    }
    else {
        password.setAttribute("type", "password");
        eye.style.color = "#808080";
    }
}



function validation() {
    var form = document.getElementById("form");
    var email = document.getElementById("email").value;
    var text = document.getElementById("text");
    var pattern = /^[^ ]+@[^ ]+\.[a-z]{2,3}$/;

    if (email.match(pattern)) {
        form.classList.add("valid");
        form.classList.remove("invalid");
        text.innerHTML = "Your Email Address is Valid.";
        text.style.color = "lightcoral";
    }
    else {
        form.classList.remove("valid");
        form.classList.add("invalid");
        text.innerHTML = "Please Enter Valid Email Address.";
        text.style.color = "lightcoral";

    }
    if (email = "") {
        form.classList.remove("valid");
        form.classList.remove("invalid");
        text.innerHTML = "";
        text.style.color = "#00ff00";
    }
}

// function popup(){
//         swal("Good job!", "You have Successfully Logged in!!", "success");
//     }


var username = document.forms["vform"]["username"];
var mailid = document.forms["vform"]["email"];
var password = document.forms["vform"]["password"];

var nameerror = document.getElementById("nameerror");
var emailerror = document.getElementById("emailerror");
var passerror = document.getElementById("passerror");
function popup() {

    if (username.value == "") {
        username.style.border = "1px solid red";
        nameerror.textContent = "Username is required";
        nameerror.style.color = "red";
        username.focus();
        return false;
    }
    if (email.value == "") {
        email.style.border = "1px solid red";
        emailerror.textContent = "EmailId is required";
        emailerror.style.color = "red";
        email.focus();
        return false;
    }
    if (password.value == "") {
        password.style.border = "1px solid red";
        passerror.textContent = "Password is required";
        passerror.style.color = "red";
        password.focus();
        return false;
    }
}

