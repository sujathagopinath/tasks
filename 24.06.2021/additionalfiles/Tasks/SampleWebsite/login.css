body{
    margin: 0;
    padding: 0;
    background: url(./images/cover.jpg);
    background-size: cover;
    background-position: center;
    font-family: sans-serif;
}
.loginbox{
    width: 340px;
    height: 550px;
    background: #000;
    color: #fff;
    top: 55%;
    left: 50%;
    position: absolute;
    transform: translate(-50%, -50%);
    box-sizing: border-box;
    padding: 70px 30px;
}

.avatar{
    width: 100px;
    height: 100px;
    border-radius: 50%;
    position: absolute;
    top: -50px;
    left: calc(50% - 50px);
}

#invalid{
    margin-bottom: 20px;
    padding: 0px;
    background: #dcdcdc;
    text-align: center;
    font-size: 14px;
    
}

h1{
    margin: 0;
    padding: 0 0 20px;
    text-align: center;
    font-size: 22px;
}
.loginbox p{
    margin: 0;
    padding: 0;
    font-weight: bold;
}
.loginbox input{
    width: 100%;
    margin-bottom: 20px;
}

.loginbox input[type="text"] 
{
    border: none;
    border-bottom: 1px solid #fff;
    background: transparent;
    outline: none;
    height: 40px;
    color: #ffff;
    font-size: 16px;
}

.loginbox input[type="email"]{
    border: none;
    border-bottom: 1px solid #fff;
    background: transparent;
    outline: none;
    height: 40px;
    color: #ffff;
    font-size: 16px;
}

.loginbox .wrapper .container input[type="password"] 
{
    border: none;
    border-bottom: 1px solid #fff;
    background: transparent;
    outline: none;
    height: 40px;
    color: #ffff;
    font-size: 16px;
    width:88%;
}

.loginbox input[type="submit"]
{
    border: none;
    outline: none;
    height: 40px;
    background: #fb2525;
    color: #fff;
    font-size: 18px;
    border-radius: 20px;
}

.loginbox input[type="submit"]:hover{
    cursor: pointer;
    background: #ffc107;
    color: #000;
}


.loginbox a{
    text-decoration: none;
    font-size: 12px;
    line-height: 20px;
    color: darkgrey;
}
.loginbox a:hover{
    color: #ffc107;
}

#text{
    display: block;
    color: #000;
    font-weight: 300;
    padding: 5px;
    font-style: italic;
}

.container{
    width: 300px;
    position: relative;
    margin: auto;
}


#toggle{
    position: absolute;
    top: 17px;
    right: 15px;
    color: #808080;
    cursor: pointer;
}
.strength{
    width: 25%;
    display: inline-block;
    position: relative;
    height: 100%;
    bottom: 5px;
}
#strength-bar{
    background-color: #dcdcdc;
    height: 10px;
    position: relative;
    width: 90%;
}
p{
    width: 100%;
    margin-top: 20px;
}
.fas{
    padding-right: 25px;
    display: inline-block;
    padding-top: 20px;
    color: #fb2525;
}

/*.fas :hover{
    color: #ffc107;
}*/