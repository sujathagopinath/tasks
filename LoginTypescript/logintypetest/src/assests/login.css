.container{
    border: 1px solid black;
    margin-bottom: 20px

}

.floating__label {
  display: block;
  position: relative;
  max-height: 0;
  margin-right: 1000px;
  font-weight: 500;
  pointer-events: none;
}


input[type=text]{
    width:50%;
    border:2px solid #aaa;
    border-radius:4px;
    margin:8px 0;
    outline:none;
    padding:8px;
    box-sizing:border-box;
    transition:.3s;
  }
  
  input[type=text]:focus{
    border-color:dodgerBlue;
    box-shadow:0 0 8px 0 dodgerBlue;
  }

  input[type=password]{
    width:50%;
    border:2px solid #aaa;
    border-radius:4px;
    margin:8px 0;
    outline:none;
    padding:8px;
    box-sizing:border-box;
    transition:.3s;
  }
  
  input[type=password]:focus{
    border-color:dodgerBlue;
    box-shadow:0 0 8px 0 dodgerBlue;
  }

  input[type=checkbox]{
    width:50%;
    border:2px solid #aaa;
    border-radius:4px;
    margin:8px 0;
    outline:none;
    padding:8px;
    box-sizing:border-box;
    transition:.3s;
  }
  
  input[type=checkbox]:focus{
    border-color:dodgerBlue;
    box-shadow:0 0 8px 0 dodgerBlue;
  }

  .button {
  background: var(--color__accent);
  border-width: 0;
  color: rgb(110, 21, 184);
  display: block;
  width: 100%;
  padding: 1rem 2rem;
  font-weight: 600;
  font-size: 1.25rem;
  text-transform: uppercase;
  letter-spacing: 1px;
}

.input-form {
  background-color: #fff;
  padding: 2rem 3rem 4rem;
  box-shadow: 2px 2px 16px 3px rgba(0, 0, 0, 0.4);
}