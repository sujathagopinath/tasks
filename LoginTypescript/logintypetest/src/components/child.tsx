import React, { FormEvent, ChangeEvent } from "react";

type SubmitEvent = FormEvent<HTMLFormElement>;
type InputEvent = ChangeEvent<HTMLInputElement>;

export interface UserProps {
  state: string;
  setState: (val: string) => void;
  handleOnSubmit: (e: SubmitEvent) => void;
  placeholder: string;
};

const Childform = (props: UserProps) => {
  return (
    <form onSubmit={props.handleOnSubmit}>
      <input
        type="text"
        name="name"
        data-testid="output"
        value={props.state}
        onChange={(e: InputEvent) => props.setState(e.target.value)}
        placeholder={props.placeholder}
      />
      <button type="submit">Submit</button>
    </form>
  );
};

export default Childform;
