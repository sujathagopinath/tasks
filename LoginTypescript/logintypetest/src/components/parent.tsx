import React, { useState } from "react";

import Childform from "./child";
type FormEvent = React.FormEvent<HTMLFormElement>;
const Parentform = () => {
  const [state, setState] = useState<string>("");
  const handleOnSubmit = (e: FormEvent) => {
    e.preventDefault();
    console.log({ state });
  };
  return (
    <Childform
      state={state}
      setState={setState}
      handleOnSubmit={handleOnSubmit}
      placeholder="Type some letters"
      data-testid="output"
    />
  );
};

export default Parentform;
