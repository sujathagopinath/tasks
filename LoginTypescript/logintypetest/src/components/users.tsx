import React, { useState } from "react";
import axios from "axios";

type User = {
  id: number;
  name: string;
};

export const Users = () => {
  const [users, setUsers] = useState<User[] | undefined>(undefined);
  const getData = () => {
    axios
      .get("https://jsonplaceholder.typicode.com/users")
      .then((response) => {
        let resUsers: User[] = response.data.map((resUser: any) => {
          return {
            id: resUser.id,
            name: resUser.name,
          };
        });
        setUsers(resUsers);
      })
      .catch((error) => {
        console.log(error);
      });
  };
  return (
    <div>
      <header>
        <p>Users Data</p>
        <button onClick={() => getData()}>Get Data</button>
        <ul>
          {users?.map((user: User) => {
            return <li key={user.id}>{user.name}</li>;
          })}
        </ul>
      </header>
    </div>
  );
};
