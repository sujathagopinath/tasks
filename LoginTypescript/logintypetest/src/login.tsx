import * as React from "react";

interface CredentialsState {
  userName: string;
  password: string;
  isLoggedIn: boolean;
  loginAttempted: boolean;
  checkbox: boolean;
}

interface CustomEvent {
  target: HTMLInputElement;
}

export class Login extends React.Component<{}, CredentialsState> {
  state: CredentialsState = {
    userName: "",
    password: "",
    isLoggedIn: false,
    loginAttempted: false,
    checkbox: false,
  };

  private async onSubmit(event: React.SyntheticEvent) {
    event.preventDefault();
    console.log("Click!!!!");
    console.log(`checked: ${this.state.checkbox}`);

    this.setState({
      loginAttempted: true,
      isLoggedIn: true,
    });
  }

  private setPassword(event: CustomEvent) {
    this.setState({ password: event.target.value });
  }
  private setUserName(event: CustomEvent) {
    this.setState({ userName: event.target.value });
  }
  private setCheck(event: CustomEvent) {
    this.setState({ checkbox: event.target.checked });
  }

  render() {
    let loginLabel;
    if (this.state.loginAttempted) {
      if (this.state.isLoggedIn) {
        loginLabel = <label>Login successful</label>;
      } else {
        loginLabel = <label>Login failed</label>;
      }
    }

    return (
      <div id="forms">
        <form data-testid="login-form" onSubmit={(e) => this.onSubmit(e)}>
          <label id="username-label">Name</label>
          <input
            data-testid="username-input"
            name="login"
            id="username-input"
            placeholder="Username"
            value={this.state.userName}
            onChange={(e) => this.setUserName(e)}
            type="text"
            aria-labelledby="username-label"
          />
          <br />
          <label id="password-label">Password</label>
          <input
            data-testid="password-input"
            name="password"
            placeholder="Password"
            value={this.state.password}
            onChange={(e) => this.setPassword(e)}
            type="password"
            aria-labelledby="password-label"
          />
          <label htmlFor="checkbox">Check</label>
          <input
            data-testid="check-input"
            name="checkbox"
            checked={this.state.checkbox}
            onChange={(e) => this.setCheck(e)}
            type="checkbox"
          />
          <br />
          <label id="submit-label">Submit</label>
          <input
            data-test="submit-button"
            type="submit"
            value="Login"
            aria-labelledby="submit-label"
          />
          <br />
        </form>
        {loginLabel}
      </div>
    );
  }
}
