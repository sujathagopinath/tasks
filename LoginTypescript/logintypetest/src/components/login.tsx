import * as React from "react";
import "../assests/login.css";

export interface CredentialsState {
  userName: string;
  password: string;
  isLoggedIn: boolean;
  loginAttempted: boolean;
  checkbox: boolean;
}

interface CustomEvent {
  target: HTMLInputElement;
}

export class Login extends React.Component<{}, CredentialsState> {
  state: CredentialsState = {
    userName: "",
    password: "",
    isLoggedIn: false,
    loginAttempted: false,
    checkbox: false,
  };

  async onSubmit(event: React.SyntheticEvent) {
    event.preventDefault();
    console.log("Click!!!!");
    console.log(`checked: ${this.state.checkbox}`);

    this.setState({
      loginAttempted: true,
      isLoggedIn: true,
    });
  }

  private setPassword(event: CustomEvent) {
    this.setState({ password: event.target.value });
  }
  private setUserName(event: CustomEvent) {
    this.setState({ userName: event.target.value });
  }
  private setCheck(event: CustomEvent) {
    this.setState({ checkbox: event.target.checked });
  }

  render() {
    let loginLabel;
    if (this.state.loginAttempted) {
      if (this.state.isLoggedIn) {
        loginLabel = <label>Login successful</label>;
      } else {
        loginLabel = <label>Login failed</label>;
      }
    }

    return (
      <div className="container">
        <form
          data-testid="login-form"
          className="input-form"
          onSubmit={(e) => this.onSubmit(e)}
        >
          <h1>Sign In</h1>
          <label id="username-label" className="floating__label">
            Name
          </label>
          <input
            data-testid="username-input"
            name="login"
            id="username-input"
            placeholder="Username"
            value={this.state.userName}
            onChange={(e) => this.setUserName(e)}
            type="text"
            aria-labelledby="username-label"
          />
          <label id="password-label" className="floating__label">
            Password
          </label>
          <input
            data-testid="password-input"
            name="password"
            placeholder="Password"
            value={this.state.password}
            onChange={(e) => this.setPassword(e)}
            type="password"
            aria-labelledby="password-label"
          />
          <label htmlFor="checkbox" className="floating__label">
            Check
          </label>
          <input
            data-testid="check-input"
            name="checkbox"
            checked={this.state.checkbox}
            onChange={(e) => this.setCheck(e)}
            type="checkbox"
          />
          <label id="submit-label" className="floating__label">
            Submit
          </label>
          <input
            data-test="submit-button"
            className="button"
            type="submit"
            value="Login"
          />
        </form>
        {loginLabel}
      </div>
    );
  }
}
