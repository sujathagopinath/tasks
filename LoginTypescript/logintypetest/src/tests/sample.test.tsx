import * as React from "react";
import { shallow, mount } from "enzyme";
import { Login } from "../components/login";
import { Users } from "../components/users";


it("renders the heading", () => {
  const wrapper = shallow(<Login />); //shallow to allows to test the component in isolation.
  expect(wrapper.find("h1").text()).toBe("Sign In");
});

// const user = {
//   name: "Darek",
//   password: "dariuszwojtowicz",
// };

// it("check the onClick callback", () => {
//   const onSubmit = jest.fn();
//   const DataInputComponent = mount(<Login user={user} />).find("input");
//   DataInputComponent.simulate("change", {
//     target: { value: "suja" },
//   });
//   expect(onSubmit).toHaveBeenCalledWith("suja");
// });

// let wrapper:any,

// beforeEach(() => {
//     wrapper = mount(<Login />);
// });
// ``
// it('should render one <Form>', () => {
//         expect(wrapper.find(Form)).toHaveLength(1);
//     });

// test("contains all users", () => {
//   const User = {
//     id: 1,
//     name: "Leanne Graham",
//   };
//   const wrapper = mount(<Users User={User} />);

//   // users.forEach((user) => {
//   expect(wrapper.text()).toMatch(User);
//   // });
// });

// test("text matches", () => {
//   const wrapper = mount(<p>Users Data</p>);
//   expect(wrapper.text()).toMatch("Users Data");
// });

it("renders whole component", () => {
  const component = shallow(<Login />);
  expect(component).toMatchSnapshot();
});

// test("submits username and password", () => {
//   const username = "hello";
//   const password = "world";
//   let onSubmit = jest.fn();

//   // const onSubmit = jest.spyOn(Login, "onSubmit");
//   const wrapper = mount(<Login onSubmit={onSubmit} />);

//   wrapper
//     .find({ "data-testid": "username-input" })
//     .simulate("change", { target: { value: username } });

//   wrapper
//     .find({ "data-testid": "password-input" })
//     .simulate("change", { target: { value: password } });

//   wrapper.update();
//   wrapper.find({ "data-testid": "loginForm" }).simulate("submit", {
//     preventDefault: () => {},
//   });

//   expect(onSubmit).toHaveBeenCalledTimes(1);
//   expect(onSubmit).toHaveBeenCalledWith({
//     username,
//     password,
//   });
// });

