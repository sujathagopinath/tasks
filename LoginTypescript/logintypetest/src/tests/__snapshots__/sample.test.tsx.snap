// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders whole component 1`] = `ShallowWrapper {}`;
